sips/sip-02X/sip-02X-wallet-interface.md

```diff
@@ -0,0 +1,244 @@
+Enums are serialized as humand-readable strings.
+BigInts are serialized as numbers, strings, or anything that can be parsed by the JavaScript BigInt constructor.
+Bytes are serialized as hex-encoded strings (without a 0x prefix).
+Predefined formats from previous SIPs are used where applicable.
+Addresses are serialized as Stacks c32-encoded strings.
+Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+## Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered recommendations for the wallet.
+The user/wallet may choose to ignore/override them.
+Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in settings like WebBTC (see WBIP-002) and other more Ethereum inspired domains.
+In other cases (e.g. WalletConnect), the namespace may already be given by meta-data (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+### Method Independent
+
+#### Common Definitions
+
+The following definitions can be used in multiple methods (mainly for transfer and transaction methods).
+
+`params`
+
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` (anything parseable by the BigInt constructor)
+- `nonce?`: `number | string` (anything parseable by the BigInt constructor)
+- `attachment?`: `string` hex-encoded
+- `anchoreMode?`: `'onChainOnly' | 'offChainOnly' | 'any'`
+- `postConditionMode?`: `'allow' | 'deny'`
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+- `sponsored?`: `boolean`, defaults to `false`
+- `address?`: `string` address, Stacks c32-encoded (sender or signer address), defaults to wallets current address
+- ~~`appDetails`~~ _removed_
+- ~~`onFinish`~~ _removed_
+- ~~`onCancel`~~ _removed_
+
+`where`
+
+- `PostCondition`: `string` hex-encoded
+
+### `stx_transferStx`
+
+`params`
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `amount`: `number | string` (anything parseable by the BigInt constructor)
+- `memo?`: `string`, defaults to `''`
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded
+
+### `stx_transferFt`
+
+`todo: haven't existed yet, should we add them?`
+
+### `stx_transferNft`
+
+`todo: haven't existed yet, should we add them?`
+
+### `stx_contractCall`
+
+`params`
+
+- `contractAddress`: `string` address, Stacks c32-encoded
+- `contractName`: `string`
+- `functionName`: `string`
+- `functionArgs`: `ClarityValue[]`, defaults to `[]`
+
+`where`
+
+- `ClarityValue`: `string` hex-encoded
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded
+
+### `stx_contractDeploy`
+
+`params`
+
+- `contractName`: `string`
+- `codeBody`: `string` Clarity contract code
+- `clarityVersion?`: `number`
```

Comment:
Would it be better to have it as a string to allow for semver or similar?

Reply:
Right now it's encoded as an integer (on the contract deploy side), so number should be the JS closest representation.